### DISCUSSION TITLE
Next.js with graphql subscriptions

### DISCUSSION THREAD

**Discussion by NickSurmanidze** [vercel/next.js#52406] [Help]
### Summary

Hello everyone,

I was using the next.js app with Graphql server running on `/pages/api/graphql` route but needed to add subscriptions (previously used SSE for Graphql subscriptions and switching to WebSockets) over WebSocket. Unfortunately, this required adding a custom server in front of the next.js app but causes an issue with HMR. All the code that is served by the express server is not reloaded and requires a full server restart which takes a few seconds. I would really like to avoid using Nodemon for this.

Maybe there is a good way to either run the WebSocket server in the API routes (seems like WS connection upgrade is ignored by api routes)  or use WebPack with back-end HMR only for the graphql-related files (separate from next.js)?

**Comment by JeffinJ** [comment]
how did you manage to implement this? please share.

**Reply by NickSurmanidze** [reply]
I ended up implementing a standalone backend app
